File: src/main/java/land/leets/domain/admin/presentation/dto/AdminDetailsResponse.java
package land.leets.domain.admin.presentation.dto;

import jakarta.validation.constraints.NotBlank;
import land.leets.domain.admin.domain.Admin;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class AdminDetailsResponse {
    @NotBlank
    private String name;

    public static AdminDetailsResponse from(Admin admin) {
        return new AdminDetailsResponse(admin.getName());
    }
}

File: src/main/kotlin/land/leets/domain/admin/presentation/dto/AdminDetailsResponse.kt
package land.leets.domain.admin.presentation.dto

import land.leets.domain.admin.domain.Admin

data class AdminDetailsResponse(
    val name: String
) {
    companion object {
        @JvmStatic
        fun from(admin: Admin): AdminDetailsResponse {
            return AdminDetailsResponse(admin.name)
        }
    }
}